source/utils/tokens.ts

```diff
@@ -7,3 +14,62 @@ export const getSymbolByChain = async (chain: string) => {
 
   return data.symbol.toString().toUpperCase();
 };
+
+export const getNativeTokenBalance = async (
+  accountAddress: string,
+  networkUrl: string
+) => {
+  try {
+    const provider = new ethers.providers.JsonRpcProvider(networkUrl);
+
+    const callBalance = await provider.getBalance(accountAddress);
+
+    const balance = ethers.utils.formatEther(callBalance);
+
+    const formattedBalance = lodash.floor(parseFloat(balance), 4);
+
+    return formattedBalance;
+  } catch (error) {
+    return 0;
+  }
+};
+
+export const getBalanceUpdatedToErcTokens = async (accountId: number) => {
+  try {
+    const { accounts, networks } = store.getState().vault;
+
+    const findAccount = accounts[accountId];
+
+    const updatedTokens = await Promise.all(
+      findAccount.assets.ethereum.map(async (vaultAssets: ITokenEthProps) => {
+        const provider = new ethers.providers.JsonRpcProvider(
+          networks.ethereum[vaultAssets.chainId].url
+        );
+
+        const _contract = new ethers.Contract(
+          vaultAssets.contractAddress,
+          vaultAssets.isNft ? getErc21Abi() : getErc20Abi(),
+          provider
+        );
+
+        const balanceCallMethod = await _contract.balanceOf(
+          findAccount.address
+        );
+
+        const balance = vaultAssets.isNft
+          ? Number(balanceCallMethod)
+          : `${balanceCallMethod / 10 ** Number(vaultAssets.decimals)}`;
+
+        const formattedBalance = vaultAssets.isNft
+          ? balance
+          : lodash.floor(parseFloat(balance as string), 4);
+
+        return { ...vaultAssets, balance: formattedBalance };
+      })
+    );
+
+    return updatedTokens;
+  } catch (error) {
+    return [];
```

Comment:
This catch probably should return the old balance of the assets, as RPCS tends to go down sometimes and also we need to know the error. Something similar to this:
`... catch (error){
 console.error('Pali utils: Couldn't update assets due to the following issue ', error);
....
return { ...vaultAssets }
}`

Reply:
fixed